homeassistant/components/gios/sensor.py

```diff
@@ -222,19 +222,27 @@ class GiosSensor(CoordinatorEntity[GiosDataUpdateCoordinator], SensorEntity):
 
     def __init__(
         self,
-        name: str,
+        name: str | None,
         coordinator: GiosDataUpdateCoordinator,
         description: GiosSensorEntityDescription,
     ) -> None:
         """Initialize."""
         super().__init__(coordinator)
-        self._attr_device_info = DeviceInfo(
-            entry_type=DeviceEntryType.SERVICE,
-            identifiers={(DOMAIN, str(coordinator.gios.station_id))},
-            manufacturer=MANUFACTURER,
-            name=name,
-            configuration_url=URL.format(station_id=coordinator.gios.station_id),
-        )
+        station_id = coordinator.gios.station_id
+        assert station_id is not None
+        station_name = coordinator.gios.measurement_stations[station_id].name
+        device_info_kwargs = {
+            "entry_type": DeviceEntryType.SERVICE,
+            "identifiers": {(DOMAIN, str(station_id))},
+            "manufacturer": MANUFACTURER,
+            "configuration_url": URL.format(station_id=station_id),
+        }
+        if name:
+            device_info_kwargs["name"] = name
+        else:
+            device_info_kwargs["translation_key"] = "station"
+            device_info_kwargs["translation_placeholders"] = {"station_name": station_name}
+        self._attr_device_info = DeviceInfo(**device_info_kwargs)
```

Comment:
I would move the creation of device info to the coordinator module:
```python
self.device_info = DeviceInfo(
    entry_type=DeviceEntryType.SERVICE,
    identifiers={(DOMAIN, str(gios.station_id))},
    manufacturer=MANUFACTURER,
    name=config_entry.data.get(CONF_NAME) or gios.station_name,
    configuration_url=URL.format(station_id=gios.station_id),
)
```
and only add here `self._attr_device_info = coordinator.device_info`

Reply:
I moved device info to coordinator.